components/TransLine.tsx

```diff
@@ -1,22 +1,50 @@
-import { Trans } from 'react-i18next'
+import { Trans, useTranslation } from 'react-i18next'
 import React from 'react'
 import { Shimmer } from './Shimmer'
-import { TransLineProps } from '../types/common'
+import { TransLineContent } from '../types/common'
+import getUrl from '../utils/getUrl'
 
-export const TransLine: React.FC<TransLineProps> = ({ loading, i18nKey, links = [] }) => {
+export interface TransLineProps extends TransLineContent {
+  loading: boolean
+  userArrivedFromUioMobile: boolean
+}
+
+export const TransLine: React.FC<TransLineProps> = ({
+  loading = false,
+  userArrivedFromUioMobile = false,
+  i18nKey,
+  links = [],
+}) => {
   if (loading) {
     return <Shimmer width={120} height={15} baseColor="#B6B2B2" shimColor="#656565" borderRadius={3} />
   }
 
-  let linkComponents: JSX.Element[] = []
+  const { t } = useTranslation(['common', 'claim-details', 'claim-status'])
+
+  const linkComponents: JSX.Element[] = []
   if (links && links.length > 0) {
-    // Disabling some linting rules for this line. The anchor <a> element will
-    // be interporlated by <Trans>.
-    /* eslint-disable jsx-a11y/anchor-has-content */
-    /* eslint-disable react/self-closing-comp */
-    linkComponents = links.map((link) => <a href={link} key={link}></a>)
-    /* eslint-enable jsx-a11y/anchor-has-content */
-    /* eslint-enable react/self-closing-comp */
+    for (const link of links) {
+      // Special case for UIO homepage links.
+      let href = ''
+      if (link === 'uio-home') {
```

Comment:
@lomky I'm totally open to changing this string/approach if you want. Since right now it seems like there is only one UIO url that changes based on mobile vs desktop, I made it a single special case and assumed we would refactor if/when we get more urls.

Reply:
I think its fine since this is the one exception, but I might pull it out into a function to keep this function readable?